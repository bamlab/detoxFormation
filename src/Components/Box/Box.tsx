import React, { FunctionComponent } from 'react';
import { StyleSheet, Text, View } from 'react-native';

const BOX_SIZE = 100;

interface BoxProps {
  text: string;
}

export const Box: FunctionComponent<BoxProps> = ({ text }) => {
  return (
    <View style={styles.box}>
      <Text style={styles.text}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  box: {
    width: BOX_SIZE,
    height: BOX_SIZE,
    backgroundColor: 'red',
    justifyContent: 'center',
  },
  text: {
    textAlign: 'center',
    color: 'green',
  },
});
