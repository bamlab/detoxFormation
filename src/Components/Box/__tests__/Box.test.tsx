import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import { Box } from '../Box';

describe('[Component] Box', () => {
  test('renders correctly', () => {
    const tree = render(<Box text="Hello" />);
    expect(tree).toMatchSnapshot();
  });
});
