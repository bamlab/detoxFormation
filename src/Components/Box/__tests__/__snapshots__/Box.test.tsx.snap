// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Box renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
      "height": 100,
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "green",
        "textAlign": "center",
      }
    }
  >
    Hello
  </Text>
</View>
`;
